plugins/module_utils/interfaces_assignments_utils.py

```diff
@@ -0,0 +1,493 @@
+
+        # Create the main element
+        main_element = Element(interface_assignment_dict["identifier"])
+
+        # Special handling for 'device' and 'descr'
+        SubElement(main_element, "if").text = interface_assignment_dict.get("device")
+        SubElement(main_element, "descr").text = interface_assignment_dict.get("descr")
+
+        # handle special cases
+        if getattr(self, "alias-subnet", None):
+            interface_assignment_dict["extra_attrs"]["alias-subnet"] = getattr(
+                self, "alias-subnet", None
+            )
+
+            interface_assignment_dict["extra_attrs"]["alias-address"] = getattr(
+                self, "alias-address", None
+            )
+
+        if getattr(self, "dhcp6-ia-pd-len", None):
+            interface_assignment_dict["extra_attrs"]["dhcp6-ia-pd-len"] = getattr(
+                self, "dhcp6-ia-pd-len", None
+            )
+
+        if getattr(self, "track6-interface", None):
+            interface_assignment_dict["extra_attrs"]["track6-interface"] = getattr(
+                self, "track6-interface", None
+            )
+
+        if getattr(self, "track6-prefix-id", None):
+            interface_assignment_dict["extra_attrs"]["track6-prefix-id"] = getattr(
+                self, "track6-prefix-id", None
+            )
+
+        # Serialize extra attributes
+        for key, value in interface_assignment_dict["extra_attrs"].items():
+            if (
+                key
+                in [
+                    "spoofmac",
+                    "alias-address",
+                    "alias-subnet",
+                    "dhcp6-ia-pd-len",
+                    "adv_dhcp_pt_timeout",
+                    "adv_dhcp_pt_retry",
+                    "adv_dhcp_pt_select_timeout",
+                    "adv_dhcp_pt_reboot",
+                    "adv_dhcp_pt_backoff_cutoff",
+                    "adv_dhcp_pt_initial_interval",
+                    "adv_dhcp_pt_values",
+                    "adv_dhcp_send_options",
+                    "adv_dhcp_request_options",
+                    "adv_dhcp_required_options",
+                    "adv_dhcp_option_modifiers",
+                    "adv_dhcp_config_advanced",
+                    "adv_dhcp_config_file_override",
+                    "adv_dhcp_config_file_override_path",
+                    "dhcprejectfrom",
+                    "track6-interface",
+                    "track6-prefix-id",
+                ]
+                and value is None
+            ):
+                sub_element = SubElement(main_element, key)
+            if value is None and key not in exceptions:
+                continue
+            sub_element = SubElement(main_element, key)
+            if value is True:
+                sub_element.text = "1"
+            elif value is not None:
+                sub_element.text = str(value)
+
+        return main_element
+
+    @classmethod
+    def from_ansible_module_params(cls, params: dict) -> "User":
```

Comment:
Copy past error? Shouldn't this return `InterfaceAssignment`?

```
def from_ansible_module_params(cls, params: dict) -> "InterfaceAssignment":

Reply:
You are absolutely right @rekup, I changed this line.